mobile-browser-based-version/server/run_server.js

```diff
@@ -1,28 +1,42 @@
-const express = require('express');
+const express               = require('express');
+const fs                    = require('fs');
 const { ExpressPeerServer } = require('peer');
-var topologies = require('./topologies.js')
+const topologies            = require('./topologies.js');
+const { makeId }            = require('./helpers.js');
+const { models }            = require('./models.js');
 
-var myArgs = process.argv.slice(2);
 
-const app = express();
+const myArgs = process.argv.slice(2);
 
-const topology = new topologies.BinaryTree()
-app.get('/', (req, res, next) => res.send('DeAI Server'));
-app.get('/neighbours/:id', (req, res, next) => res.send(topology.getNeighbours(req.params['id'])));
+const app = express();
 
 const server = app.listen(myArgs[0]);
-
 const peerServer = ExpressPeerServer(server, {
-    path: '/',
+    path: '/peerjs',
+    key: 'api',
     allow_discovery: true,
-  });
-app.use('/deai', peerServer);
+    generateClientId: makeId(12)
+});
 
+var topology = new topologies.BinaryTree();
 
-peerServer.on('connection', (client) => { 
+peerServer.on('connection', (client) => {
     topology.addPeer(client.getId())
 });
 
-peerServer.on('disconnect', (client) => { 
+peerServer.on('disconnect', (client) => {
     topology.removePeer(client.getId())
-});
\ No newline at end of file
+});
+
+const tasks = JSON.parse(fs.readFileSync(myArgs[1]));
+
+const tasksRouter = express.Router();
+tasksRouter.get('/', (req, res) => res.send(tasks));
+tasks.forEach(task => {
+    tasksRouter.get('/' + task.taskId, (req, res) => res.send(models.get(task.taskId)))
+});
+
+app.get('/', (req, res) => res.send('DeAI Server'));
+app.use('/', peerServer);
```

Comment:
I think this should be `app.use('/deai', peerServer);` to avoid conflicts

Reply:
This would just add an empty route, thus lengthening URLs. Currently, these two lines of code produce the following routes
- /: home page of the server
- /peerjs: home page of the PeerServer
- /peerjs/api/*: all services from the PeerServer
So there is no loss of clarity, nor any conflict with the server's home page as PeerServer doesn't serve anything on its mounting route (/).